roles/conf/distsearch/tasks/distributedSearch/server.yml

```diff
@@ -9,11 +9,14 @@
 
 # server
 
+
 - name: "Configure distsearch.conf [distributedSearch] - server state=present"
+  vars:
+    searchpeers: "{{ groups['shcmember'] }}, {{ groups['licensemaster'] }}, {{ groups['masternode'] }}"
   ini_file: dest={{ splunk_conf_path }}/distsearch.conf
             section=distributedSearch
             option=servers
-            value="{{ splunk_distsearch_conf.distributedSearch.servers | create_distsearch_servers  }}"
+            value="{{ searchpeers | create_distsearch_servers  }}"
             state=present
   when: splunk_distsearch_conf.distributedSearch.servers is defined
```

Comment:
If you use the var `splunk_distsearch_conf.distributedSearch.servers`, every group listed in `servers[]` gets set as a search peer in the distributedSearch stanza. Only the License Master, Search Cluster Members, and Index Cluster Master should be listed as Search Peers on the DMC. ([splunk docs](http://docs.splunk.com/Documentation/Splunk/6.2.8/Admin/ConfiguretheMonitoringConsole#Add_instances_as_search_peers), under "Add instances as search peers."

Listing those nodes under [searchpeers] and running configure_searchpeers.yml shouldn't have any negative effects, as it's just the web/inputs/server conf roles which already get ran by other playbooks.